seqr/views/utils/project_context_utils.py

```diff
@@ -0,0 +1,140 @@
+
+    response = {
+        'projectsByGuid': projects_by_guid,
+        'familiesByGuid': {f['familyGuid']: f for f in families},
+        'individualsByGuid': {i['individualGuid']: i for i in individuals},
+        'samplesByGuid': {s['sampleGuid']: s for s in samples},
+        'igvSamplesByGuid': {s['sampleGuid']: s for s in igv_samples},
+        'locusListsByGuid': locus_lists_by_guid,
+        'analysisGroupsByGuid': {ag['analysisGroupGuid']: ag for ag in analysis_groups},
+    }
+
+    if project_guid:
+        response['projectsByGuid'][project_guid]['locusListGuids'] = list(response['locusListsByGuid'].keys())
+    else:
+        _add_parent_ids(response, projects, family_models, individual_models, locus_lists_models)
+
+    _add_child_ids(response)
+
+    return response
+
+def _add_parent_ids(response, projects, family_models, individual_models, locus_lists_models):
+    project_id_to_guid = {project.id: project.guid for project in projects}
+    family_id_to_guid = {family.id: family.guid for family in family_models}
+    individual_id_to_guid = {individual.id: individual.guid for individual in individual_models}
+    family_guid_to_project_guid = {}
+    individual_guid_to_project_guid = {}
+    for family in response['familiesByGuid'].values():
+        project_guid = project_id_to_guid[family.pop('projectId')]
+        family['projectGuid'] = project_guid
+        family_guid_to_project_guid[family['familyGuid']] = project_guid
+    for individual in response['individualsByGuid'].values():
+        family_guid = family_id_to_guid[individual.pop('familyId')]
+        project_guid = family_guid_to_project_guid[family_guid]
+        individual['familyGuid'] = family_guid
+        individual['projectGuid'] = project_guid
+        individual_guid_to_project_guid[individual['individualGuid']] = project_guid
+    for sample in response['samplesByGuid'].values():
+        individual_guid = individual_id_to_guid[sample.pop('individualId')]
+        sample['individualGuid'] = individual_guid
+        sample['projectGuid'] = individual_guid_to_project_guid[individual_guid]
+    for sample in response['igvSamplesByGuid'].values():
+        individual_guid = individual_id_to_guid[sample.pop('individualId')]
+        sample['individualGuid'] = individual_guid
+        sample['projectGuid'] = individual_guid_to_project_guid[individual_guid]
+    for group in response['analysisGroupsByGuid'].values():
+        group['projectGuid'] = project_id_to_guid[group.pop('projectId')]
+
+    for project in response['projectsByGuid'].values():
+        project['locusListGuids'] = []
+    prefetch_related_objects(locus_lists_models, 'projects')
+    for locus_list in locus_lists_models:
+        for project in locus_list.projects.all():
+            response['projectsByGuid'][project.guid]['locusListGuids'].append(locus_list.guid)
+
+def _add_child_ids(response):
```

Comment:
Would it be more descriptive to rename it to something like `_add_individual_guids`?

Reply:
no because it is adding `sampleGuids`, and `igvSampleGuids` as well. It is adding the guids for all of the child models, so this is a much more descriptive name

Reply:
I see what it means now. All the projects, families, individuals, samples, and igvSamples in the response need both their parent models and child models' guids.